PasswordValidator_UnitTests/AdvancedValidator_UnitTests.cs

```diff
@@ -1,47 +1,150 @@
 ﻿using NUnit.Framework;
+using NSubstitute;
+using PasswordValidator.Rules;
 using PasswordValidator.Validators;
+using PasswordValidator.Enums;
 using System;
+using System.Collections.Generic;
 
 namespace PasswordValidator_UnitTests
 {
     internal class AdvancedValidator_UnitTests
     {
-        private readonly IPasswordValidator _validator;
+        private IPasswordValidator? _validator;
+        private readonly IRule _mockSimpleLengthRule;
+        private readonly IRule _mockSimpleNumberRule;
+        private readonly IRule _mockAdvancedLengthRule;
+        private readonly IRule _mockAdvancedNumberRule;
+        private readonly IRule _mockCharacterRule;
+
 
         public AdvancedValidator_UnitTests()
         {
-            _validator = new AdvancedValidator();
+            _mockCharacterRule = Substitute.For<IRule>();
+            _mockSimpleLengthRule = Substitute.For<IRule>();
+            _mockSimpleNumberRule = Substitute.For<IRule>();
+            _mockAdvancedNumberRule = Substitute.For<IRule>();
+            _mockAdvancedLengthRule = Substitute.For<IRule>();
+
+            _mockCharacterRule.Type.Returns(RuleType.ContainsSpecialCharacter);
+            _mockSimpleLengthRule.Type.Returns(RuleType.SimpleLength);
+            _mockSimpleNumberRule.Type.Returns(RuleType.SimpleNumber);
+            _mockAdvancedLengthRule.Type.Returns(RuleType.AdvancedLength);
+            _mockAdvancedNumberRule.Type.Returns(RuleType.AdvancedNumber);
         }
 
-        [TestCase("aieinr343_3@")]
-        [TestCase("111dfjiea@e3")]
-        [TestCase("3jda;af4fdaffede4")]
-        [TestCase("qqqwww333@@@")]
-        [TestCase("f92-v;w/2wqfas")]
-        [TestCase("29fjvie[w'fwfk")]
-        public void IsValidAdvancedPassword_ValidPassword_ShouldReturnTrue(string advancedPassword)
+        [TearDown]
+        public void CleanUp()
         {
-            var result = _validator.Validate(advancedPassword);
+            _mockCharacterRule.ClearReceivedCalls();
+            _mockSimpleLengthRule.ClearReceivedCalls();
+            _mockSimpleNumberRule.ClearReceivedCalls();
+            _mockAdvancedLengthRule.ClearReceivedCalls();
+            _mockAdvancedNumberRule.ClearReceivedCalls();
+        }
+
+        [Test]
+        public void Validate_AllRulesSatisfied_ShouldReturnTrue()
+        {
+            _mockAdvancedLengthRule.Satisfied(Arg.Any<string>()).Returns(true);
+            _mockAdvancedNumberRule.Satisfied(Arg.Any<string>()).Returns(true);
+            _mockCharacterRule.Satisfied(Arg.Any<string>()).Returns(true);
+
+            var mockRules = new IRule[] { _mockAdvancedLengthRule, _mockAdvancedNumberRule, _mockCharacterRule, _mockSimpleLengthRule, _mockSimpleNumberRule };
+            _validator = new AdvancedValidator(mockRules);
+
+            var result = _validator.Validate("testPassword");
+
             Assert.IsTrue(result);
+            _mockAdvancedLengthRule.Received().Satisfied(Arg.Any<string>());
```

Comment:
Again - replace Arg.Any<string> with the input you expect to receive - that way you will know things are 'correct' - for example it could pass "abc" in not "testPassword"